hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/ReconTaskControllerImpl.java

```diff
@@ -227,6 +265,32 @@ public synchronized void reInitializeTasks(ReconOMMetadataManager omMetadataMana
     } catch (CancellationException ce) {
       LOG.error("Some tasks were cancelled with exception", ce);
     }
+
+    if (isRunSuccessful.get()) {
+      try {
+        reconDBProvider.replaceStagedDb(stagedReconDBProvider);
+        reconNamespaceSummaryManager.reinitialize(reconDBProvider);
+        reconContainerMetadataManager.reinitialize(reconDBProvider);
+        reprocessTaskStatus.setLastTaskRunStatus(0);
+        reprocessTaskStatus.recordRunCompletion();
+        LOG.info("Re-initialization of tasks completed successfully.");
+      } catch (Exception e) {
+        LOG.error("Re-initialization of tasks failed.", e);
+        reprocessTaskStatus.setLastTaskRunStatus(-1);
+        reprocessTaskStatus.recordRunCompletion();
+        // reinitialize the Recon OM tasks with the original DB provider
+        try {
+          reconNamespaceSummaryManager.reinitialize(reconDBProvider);
+          reconContainerMetadataManager.reinitialize(reconDBProvider);
+        } catch (IOException ex) {
+          LOG.error("Re-initialization of task manager failed.", e);
+        }
+      }
+    } else {
+      LOG.error("Reprocess of task has failed.");
+      reprocessTaskStatus.setLastTaskRunStatus(-1);
+      reprocessTaskStatus.recordRunCompletion();
```

Comment:
Sameway here also , not just STAGING DB switchover task status to be updated here, but also need to update the status of individual task reprocess status as well.

Reply:
updated